mobile/src/events/DatePicker.tsx

```diff
@@ -0,0 +1,132 @@
+import { useField, useFormikContext } from "formik";
+import React, { useState } from "react";
+import {
+  View,
+  Text,
+  Platform,
+  StyleSheet,
+} from "react-native";
+import DateTimePicker from '@react-native-community/datetimepicker';
+
+const DatePicker: React.FC<{}> = ({ ...props }) => {
+  // Used for user interaction with datepicker
+  const [date, setDate] = useState(props.date);
+  const [mode, setMode] = useState('date');
+  const [show, setShow] = useState(false);
+
+  // Used with setting date value in formik 
+  const { setFieldValue } = useFormikContext();
+  const [field] = useField(props.name);
+
+  const onChange = (event, selectedDate) => {
+    // Set date on calendar UI
+    setDate(selectedDate);
+    // Set date in form 
+    setFieldValue(props.name, selectedDate);
+    console.log("Selected date", selectedDate);
+  };
+
+  const showMode = (currentMode) => {
+    setShow(true);
+    setMode(currentMode);
+  };
+
+  const showDatepicker = () => {
+    showMode('date');
+  };
+
+  const showTimepicker = () => {
+    showMode('time');
+  };
+
+  return (
+    <View>
+      <View style={styles.dateTimeRow}>
+        {/* // Date  */}
+        <Text onPress={showDatepicker}>
+          {dayNumToString(date.getDay())}, {monthNumToString(date.getMonth())} {date.getDate()}
+        </Text>
+        {/* // Time  */}
+        <Text onPress={showTimepicker}>   {date.toLocaleTimeString('en-US')}</Text>
+      </View>
+
+      {show && (
+        <DateTimePicker
+          {...field}
+          {...props}
+          testID="dateTimePicker"
+          value={date}
+          mode={mode}
+          is24Hour={true}
+          display="default"
+          onChange={onChange}
+        />
+      )}
+    </View>
+  );
+};
+
+const dayNumToString = (day: number) => {
```

Comment:
Not sure if this is relevant anymore since you already wrote this, but moment.js does all this parsing stuff for you.

Reply:
MOMENT ROCKS. Changed it